src/Dict.h

```diff
@@ -2,79 +2,205 @@
 
 #pragma once
 
-#include "zeek-config.h"
+#include <cstdint>
+#include <vector>
 
-#include "List.h"
 #include "Hash.h"
 
-ZEEK_FORWARD_DECLARE_NAMESPACED(DictEntry, zeek::detail);
 ZEEK_FORWARD_DECLARE_NAMESPACED(IterCookie, zeek);
+ZEEK_FORWARD_DECLARE_NAMESPACED(DictEntry, zeek::detail);
 
 // Type for function to be called when deleting elements.
 typedef void (*dict_delete_func)(void*);
 
 namespace zeek {
 
-// Type indicating whether the dictionary should keep track of the order
-// of insertions.
+// Default number of hash buckets in dictionary.  The dictionary will increase the size
+// of the hash table as needed.
+constexpr uint32_t HASH_MASK = 0xFFFFFFFF; //only lower 32 bits.
+
+// These four variables can be used to build different targets with -Dxxx for performance
+// or for debugging purposes.
+
+// When incrementally resizing and remapping, it remaps DICT_REMAP_ENTRIES each step. Use
+// 2 for debug. 16 is best for a release build.
+#ifndef DICT_REMAP_ENTRIES
```

Comment:
I'd take out the `#ifndef`, very unlikely anybody will ever want to change set any of them; and if so, can always patch the source.